helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -431,9 +426,104 @@ public Response getClusterStateModelDefinition(@PathParam("clusterId") String cl
     StateModelDefinition stateModelDef =
         dataAccessor.getProperty(dataAccessor.keyBuilder().stateModelDef(statemodel));
 
+    if (stateModelDef == null) {
+      return  notFound();
+    }
     return JSONRepresentation(stateModelDef.getRecord());
   }
 
+  @PUT
+  @Path("{clusterId}/statemodeldefs/{statemodel}")
+  public Response createClusterStateModelDefinition(
+      @PathParam("clusterId") String clusterId,
+      @PathParam("statemodel") String statemodel,
+      String content
+  ) {
+    ZNRecord record;
+    try {
+      record = toZNRecord(content);
+    } catch (IOException e) {
+      _logger.error("Failed to deserialize user's input " + content + ", Exception: " + e);
+      return badRequest("Input is not a valid ZNRecord!");
+    }
+    HelixZkClient zkClient = getHelixZkClient();
+    String path = PropertyPathBuilder.stateModelDef(clusterId);
+    try {
+      ZKUtil.createChildren(zkClient, path, record);
+    } catch (Exception e) {
```

Comment:
If an exception needs to caught, does this exception need to be more specific? Do we know what potential exception is thrown, like InvalidACLException/InterruptedException/ZkNodeExistsException?

Reply:
There could be other exceptions not necessarily as InvalidACLException/InterruptedException/ZkNodeExistsException I guess. The idea is to catch them and log them instead of let the framework handle them, which they would return 500, instead of badRequest, which give an idea as unable to create the znode.

Reply:
I get you.

Since it is a znode creation, I guess all possible exceptions may be included in KeeperException  https://zookeeper.apache.org/doc/r3.3.2/api/org/apache/zookeeper/KeeperException.html

Maybe we could enhance HelixException to support all the potential exceptions?

Reply:
That is probably beyond the scope of this diff. How about let us keep it this way for now.